worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongePlatform.java

```diff
@@ -0,0 +1,190 @@
+import com.sk89q.worldedit.sponge.config.SpongeConfiguration;
+import com.sk89q.worldedit.sponge.nms.IDHelper;
+import com.sk89q.worldedit.sponge.nms.SpongeNMSWorld;
+import com.sk89q.worldedit.util.command.CommandMapping;
+import com.sk89q.worldedit.util.command.Dispatcher;
+import com.sk89q.worldedit.world.World;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.entity.EntityType;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.scheduler.Task;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+class SpongePlatform extends AbstractPlatform implements MultiUserPlatform {
+
+    private final SpongeWorldEdit mod;
+    private boolean hookingEvents = false;
+
+    SpongePlatform(SpongeWorldEdit mod) {
+        this.mod = mod;
+    }
+
+    boolean isHookingEvents() {
+        return hookingEvents;
+    }
+
+    @Override
+    public int resolveItem(String name) {
+        if (name == null) return 0;
+
+        Optional<ItemType> optType = Sponge.getRegistry().getType(ItemType.class, name);
+
+        if (optType.isPresent()) {
+            return IDHelper.resolve(optType.get());
+        }
+
+        return 0;
+    }
+
+    @Override
+    public boolean isValidMobType(String type) {
+        return Sponge.getRegistry().getType(EntityType.class, type).isPresent();
+    }
+
+    @Override
+    public void reload() {
+        getConfiguration().load();
+    }
+
+    @Override
+    public int schedule(long delay, long period, Runnable task) {
+        Task.builder().delayTicks(delay).intervalTicks(period).execute(task).submit(SpongeWorldEdit.inst());
+        return 0; // TODO This isn't right, but we only check for -1 values
+    }
+
+    @Override
+    public List<? extends com.sk89q.worldedit.world.World> getWorlds() {
+        Collection<org.spongepowered.api.world.World> worlds = Sponge.getServer().getWorlds();
+        List<com.sk89q.worldedit.world.World> ret = new ArrayList<>(worlds.size());
+        for (org.spongepowered.api.world.World world : worlds) {
+            ret.add(new SpongeNMSWorld(world));
+        }
+        return ret;
+    }
+
+    @Nullable
+    @Override
+    public Player matchPlayer(Player player) {
+        if (player instanceof SpongePlayer) {
+            return player;
+        } else {
+            Optional<org.spongepowered.api.entity.living.player.Player> optPlayer = Sponge.getServer().getPlayer(player.getUniqueId());
+            return optPlayer.isPresent() ? new SpongePlayer(this, optPlayer.get()) : null;
+        }
+    }
+
+    @Nullable
+    @Override
+    public World matchWorld(World world) {
+        if (world instanceof SpongeWorld) {
+            return world;
+        } else {
+            for (org.spongepowered.api.world.World ws : Sponge.getServer().getWorlds()) {
+                if (ws.getName().equals(world.getName())) {
```

Comment:
We don't currently provide a concept of a world UUID in our abstraction:
https://github.com/sk89q/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/world/World.java